packages/core/lib/engine_http.js

```diff
@@ -399,12 +401,22 @@ HttpEngine.prototype.step = function step(requestSpec, ee, opts) {
           requestParams.body = _.reduce(
             requestParams.formData,
             function (acc, v, k) {
-              // acc[k] = template(v, context);
-              acc.append(k, template(v, context));
+              let V = template(v, context);
+              if (V && _.isPlainObject(V) && V.fromFile) {
+                V = fs.createReadStream(
+                  path.resolve(process.cwd(), V.fromFile)
+                );
+              }
+              acc.append(k, V);
               return acc;
             },
             f
           );
+          if (params.setContentLengthHeader) {
+            requestParams.headers = requestParams.headers || {};
+            requestParams.headers['content-length'] =
+              requestParams.body.getLengthSync();
```

Comment:
This doesn't work - https://github.com/form-data/form-data?tab=readme-ov-file#integer-getlengthsync

Reply:
Resolved using `fs` like plugin does. Commit https://github.com/artilleryio/artillery/pull/3232/commits/b7e07115bbf34c9513ac21bc88101c6603f2aca9